internal/cluster-object/namespace_provider_test.go

```diff
@@ -0,0 +1,71 @@
+package clusterobject
+
+import (
+	"context"
+	"log/slog"
+	"os"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestNamespaceProvider(t *testing.T) {
+	// given
+	logger := slog.New(slog.NewTextHandler(os.Stdout, nil))
+
+	t.Run("should fetch all namespaces", func(t *testing.T) {
+		// given
+		namespaces := initNamespaces()
+		k8sClient := fake.NewClientBuilder().WithLists(namespaces).Build()
+		nsProvider := NewNamespaceProvider(k8sClient, logger)
+
+		// when
+		nsList, err := nsProvider.All(context.TODO())
+
+		// then
+		if err != nil {
+			t.Errorf("Error while fetching namespaces: %s", err)
+		}
+		if len(nsList.Items) != 3 {
```

Comment:
Can we use assert specific value?

Reply:
Sure, I will replace if with assert.Len()